Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java

```diff
@@ -125,6 +131,33 @@ private void updateSteerAngle() {
         }
     }
 
+    private void updateBoost() {
+
+        // Only allow boosting when the trail is gone.
+        if (car.getThrottleAction() == ThrottleAction.BOOST && trail.size() == 0) {
+            boostActive = true;
+        }
+
+        // Disable boost when the throttle action is not boost or boost amount is 0
+        if (car.getThrottleAction() != ThrottleAction.BOOST || car.getBoostAmount() == 0) {
+            boostActive = false;
+        }
+
+        // The trail has to remove itself when it's longer than max length or boost is not active.
+        if (trail.size() > PhysicsConstants.CAR_BOOST_TRAIL_SIZE || trail.size() > 0 && !boostActive) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1155)